source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRTextView.mm

```diff
@@ -0,0 +1,95 @@
+//
+//  ACRTextView
+//  ACRTextView.mm
+//
+//  Copyright © 2017 Microsoft. All rights reserved.
+//
+
+#import "ACOBaseCardElementPrivate.h"
+#import "ACRTextView.h"
+#import "TextInput.h"
+
+@implementation ACRTextView
+
+- (instancetype)initWithFrame:(CGRect)frame element:(ACOBaseCardElement *)element
+{
+    self = [super initWithFrame:frame];
+    if (self) {
+        std::shared_ptr<BaseCardElement> elem = [element element];
+        std::shared_ptr<TextInput> inputBlck = std::dynamic_pointer_cast<TextInput>(elem);
+        _maxLength = inputBlck->GetMaxLength();
+        _placeholderText = [[NSString alloc] initWithCString:inputBlck->GetPlaceholder().c_str() encoding:NSUTF8StringEncoding];
+        if(inputBlck->GetValue().size()){
+            self.text = [[NSString alloc] initWithCString:inputBlck->GetValue().c_str() encoding:NSUTF8StringEncoding];
+        } else if([_placeholderText length]){
+            self.text = _placeholderText;
+            self.textColor = [UIColor lightGrayColor];
+        }
+    }
+    return self;
+}
+
+- (BOOL)validate:(NSError **)error
+{
+    if (self.isRequired && !self.hasText) {
+        if (error) {
+            *error = [NSError errorWithDomain:ACRInputErrorDomain code:ACRInputErrorValueMissing userInfo:nil];
+        } return NO;
```

Comment:
>  [](start = 9, length = 1)

Newline

Reply:
thanks!

---
In reply to: [198291703](https://github.com/Microsoft/AdaptiveCards/pull/1643#discussion_r198291703) [](ancestors = 198291703)